src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -282,7 +286,8 @@ SHAMapStoreImp::run()
     while (1)
     {
         healthy_ = true;
-        validatedLedger_.reset();
+        validatedLedger.reset();
```

Comment:
@ximinez I changed my mind about using a lambda here. The lambda is doing much more than just complex initialization. It's both changing the object's member vars and it effecting control flow of the caller (through the `bool` part of the pair return value). We're better off with the the straight-ahead non-lambda code here.

Sorry for changing my mind, and I'm OK with it if you still prefer the lambda.